config/routes.rb

```diff
@@ -77,7 +77,11 @@
         json_api_resources :pages, controller: "project_pages"
       end
 
-      json_api_resources :workflows, links: [:subject_sets], versioned: true
+      json_api_resources :workflows, links: [:subject_sets], versioned: true do
+        member do
+          post :retire_subject
```

Comment:
Why not make this is a JSON API relationship called `retired_subjects` and retire them by posting to `/workflows/1/links/retired_subjects` would also all for easy 'unretiring' by doing deleting `/workflows/1/links/retired_subjects/1,11,2`